### PR TITLE
[feature] add ability to set global filter/sort/display for Manga chapters

### PR DESCRIPTION
**Pull request by lmj0011**
closes https://github.com/inorichi/tachiyomi/issues/484

This PR adds functionality to allow setting global default "Chapter Settings":

- The behavior after setting default "Chapter Settings" is that it applies to all Manga being browsed or _newly_ added to your Library

demo: https://streamable.com/3ntsrp

- To set default "Chapter Settings" for all Manga in your Library, you must long-press the "Set as Default" button and then tick the "Apply to all in Library" checkbox first, before tapping "Set as Default"

demo: https://streamable.com/6b9z5l

### CODE REVIEW

File: app/src/main/java/eu/kanade/tachiyomi/data/preference/PreferencesHelper.kt [hunk 1]
```diff
@@ -244,4 +245,32 @@ class PreferencesHelper(val context: Context) {
...
+    fun filterChapterByBookmarked() = prefs.getInt(Keys.filterChapterByBookmarked, Manga.SHOW_ALL)
+
+    fun sortChapterBySourceOrNumber() = prefs.getInt(Keys.sortChapterBySourceOrNumber, Manga.SORTING_SOURCE)
+
+    fun displayChapterByNameOrNumber() = prefs.getInt(Keys.displayChapterByNameOrNumber, Manga.DISPLAY_NAME)
+
+    fun sortChapterByAscendingOrDescending() = prefs.getInt(Keys.sortChapterByAscendingOrDescending, Manga.SORT_DESC)
+
+    fun setChapterSettingsDefault(m: Manga) {
+        val editor = prefs.edit()
```
[the remark this answers is not in the document]
Comment (lmj0011) [reply app/src/main/java/eu/kanade/tachiyomi/data/preference/PreferencesHelper.kt]: see f754f9686f3d86d17e67f3b05c6a10d09a089d5f

File: app/src/main/java/eu/kanade/tachiyomi/data/preference/PreferenceKeys.kt [hunk 2]
```diff
@@ -161,6 +161,18 @@ object PreferenceKeys {
 
     const val enableDoh = "enable_doh"
 
+    const val filterChapterByRead = "filter_chapter_by_read"
```
[the remark this answers is not in the document]
Comment (lmj0011) [reply app/src/main/java/eu/kanade/tachiyomi/data/preference/PreferenceKeys.kt]: see f754f9686f3d86d17e67f3b05c6a10d09a089d5f

File: app/src/main/java/eu/kanade/tachiyomi/util/chapter/ChapterSettingsHelper.kt [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        return when {
+            (
+                m.readFilter == Manga.SHOW_ALL &&
+                    m.downloadedFilter == Manga.SHOW_ALL &&
+                    m.bookmarkedFilter == Manga.SHOW_ALL &&
+                    m.sorting == Manga.SORTING_SOURCE &&
+                    m.displayMode == Manga.DISPLAY_NAME &&
+                    sort == Manga.SORT_DESC
+                ) -> true
+            else -> false
```
[the remark this answers is not in the document]
Comment (lmj0011) [reply app/src/main/java/eu/kanade/tachiyomi/util/chapter/ChapterSettingsHelper.kt]: see f754f9686f3d86d17e67f3b05c6a10d09a089d5f

File: app/src/main/java/eu/kanade/tachiyomi/widget/TabbedBottomSheetDialog.kt [hunk 4]
```diff
@@ -4,25 +4,49 @@ import android.app.Activity
...
+        }
+
         setContentView(binding.root)
     }
 
     abstract fun getTabViews(): List<View>
 
     abstract fun getTabTitles(): List<Int>
 
+    fun toggleSaveChapterSettingsBtnVisibility(m: Manga) {
```
[the remark this answers is not in the document]
Comment (lmj0011) [reply app/src/main/java/eu/kanade/tachiyomi/widget/TabbedBottomSheetDialog.kt]: see f754f9686f3d86d17e67f3b05c6a10d09a089d5f

**Comment by lmj0011** [comment]
> Regarding the UX, I'm not really a fan of how the button suddenly appears/the apply all functionality is hidden.
> 
> Maybe we can have an overflow menu at the top right of the sheet, slightly offsetting the tabs?

this is how it looks now https://streamable.com/gvzqv5

**Comment by arkon** [comment]
Rather than the checkbox being there, how about a dialog that appears when you apply it with a checkbox in there, similar to removing items from the library also prompting to delete downloaded chapters?

**Comment by lmj0011** [comment]
> Rather than the checkbox being there, how about a dialog that appears when you apply it with a checkbox in there, similar to removing items from the library also prompting to delete downloaded chapters?

https://github.com/inorichi/tachiyomi/pull/3622/commits/98fd358a06ededcaa4ef4f7be5a44291daff0d04

**Pull request merged**